editor/scene_tree_dock.cpp

```diff
@@ -1634,50 +1634,61 @@ bool SceneTreeDock::_update_node_path(Node *p_root_node, NodePath &r_node_path,
 	return false;
 }
 
-bool SceneTreeDock::_check_node_path_recursive(Node *p_root_node, Variant &r_variant, HashMap<Node *, NodePath> *p_renames) const {
+bool SceneTreeDock::_check_node_path_recursive(Node *p_root_node, Variant &r_variant, Variant &r_cleared_variant, HashMap<Node *, NodePath> *p_renames) const {
 	switch (r_variant.get_type()) {
 		case Variant::NODE_PATH: {
 			NodePath node_path = r_variant;
 			if (!node_path.is_empty() && _update_node_path(p_root_node, node_path, p_renames)) {
 				r_variant = node_path;
+				r_cleared_variant = NodePath();
 				return true;
 			}
 		} break;
 
 		case Variant::ARRAY: {
 			Array a = r_variant;
+			Array cleared_a = r_variant;
 			bool updated = false;
 			for (int i = 0; i < a.size(); i++) {
 				Variant value = a[i];
-				if (_check_node_path_recursive(p_root_node, value, p_renames)) {
+				Variant cleared_value;
+				if (_check_node_path_recursive(p_root_node, value, cleared_value, p_renames)) {
 					if (!updated) {
 						a = a.duplicate(); // Need to duplicate for undo-redo to work.
+						cleared_a = cleared_a.duplicate();
 						updated = true;
 					}
 					a[i] = value;
+					cleared_a[i] = cleared_value;
 				}
 			}
 			if (updated) {
 				r_variant = a;
+				r_cleared_variant = cleared_a;
 				return true;
 			}
 		} break;
 
 		case Variant::DICTIONARY: {
 			Dictionary d = r_variant;
+			Dictionary cleared_d = r_variant;
 			bool updated = false;
 			for (int i = 0; i < d.size(); i++) {
 				Variant value = d.get_value_at_index(i);
-				if (_check_node_path_recursive(p_root_node, value, p_renames)) {
+				Variant cleared_value;
+				if (_check_node_path_recursive(p_root_node, value, cleared_value, p_renames)) {
 					if (!updated) {
 						d = d.duplicate(); // Need to duplicate for undo-redo to work.
+						cleared_d = cleared_d.duplicate();
 						updated = true;
 					}
 					d[d.get_key_at_index(i)] = value;
+					cleared_d[d.get_key_at_index(i)] = cleared_value;
```

Comment:
`get_key_at_index()` should be stored in some variable, to avoid fetching it twice.